crates/lair_keystore_api/src/lair_server/priv_api.rs

```diff
@@ -233,6 +233,106 @@ pub(crate) fn priv_req_new_seed<'a>(
         Ok(())
     }
 }
+pub(crate) fn priv_req_derive_seed<'a>(
+    inner: &'a Arc<RwLock<SrvInner>>,
+    send: &'a crate::sodium_secretstream::S3Sender<LairApiEnum>,
+    dec_ctx_key: &'a sodoken::BufReadSized<32>,
+    unlocked: &'a Arc<atomic::AtomicBool>,
+    req: LairApiReqDeriveSeed,
+) -> impl Future<Output = LairResult<()>> + 'a + Send {
+    async move {
+        let store = priv_get_store(inner, unlocked)?;
+
+        let entry = store.get_entry_by_tag(req.src_tag.clone()).await?;
+
+        let (src_seed, src_seed_info) = match (&*entry, req.src_deep_lock_passphrase) {
+            (LairEntryInner::Seed { seed, seed_info, .. }, None) => {
+                let seed = seed.decrypt(store.get_bidi_ctx_key()).await?;
+                (seed, seed_info)
+            }
+
+            (
+                LairEntryInner::DeepLockedSeed {
+                    seed,
+                    seed_info,
+                    salt,
+                    ..
+                },
+                Some(secret),
+            ) => {
+                let deep_key = deep_unlock_key_from_passphrase(
+                    &secret,
+                    dec_ctx_key,
+                    BufReadSized::from(salt.clone().cloned_inner()),
+                )
+                .await?;
+                let seed = seed.decrypt(deep_key).await?;
+                (seed, seed_info)
+            },
+
+            (LairEntryInner::WkaTlsCert { .. }, _) => return Err("The tag provided is for a Cert, which cannot be derived. You must specify the tag for a Seed.".into()),
+            (LairEntryInner::Seed { .. }, Some(_)) => return Err("A passphrase was provided for a seed which is not deep-locked. Make the request without a `src_passphrase`.".into()),
+            (LairEntryInner::DeepLockedSeed { .. }, None) => return Err("A `src_passphrase` is needed to unlock the source seed which is deep-locked.".into()),
+        };
+
+        let mut parent = src_seed;
+
+        for index in req.derivation_path.iter() {
+            let derived = BufWriteSized::new_mem_locked()?;
+            sodoken::kdf::derive_from_key(
+                derived.clone(),
+                *index as u64,
+                *b"IdnSecKy",
```

Comment:
Whoops, sorry I missed this earlier, this has to match the seed bundle spec.

```suggestion
                *b"SeedBndl",
```